pkg/cmd/roachtest/tests/validate_grant_option.go

```diff
@@ -0,0 +1,134 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/util/version"
+	"github.com/stretchr/testify/require"
+)
+
+func registerValidateGrantOption(r registry.Registry) {
+	r.Add(registry.TestSpec{
+		Name:    "sql-experience/validate-grant-option",
+		Owner:   registry.OwnerSQLExperience,
+		Cluster: r.MakeClusterSpec(3),
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			runRegisterValidateGrantOption(ctx, t, c, *t.BuildVersion())
+		},
+	})
+}
+
+// execSQL executes the SQL statement as "root".
+func execSQL(sqlStatement string, errorExpected bool, node int) versionStep {
+	return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
+		conn, err := u.c.ConnE(ctx, node)
+		require.NoError(t, err)
+		_, err = conn.Exec(sqlStatement)
+		if !errorExpected {
+			require.NoError(t, err)
+		} else {
+			require.Error(t, err)
+		}
+	}
+}
+
+// execSQLUser changes the current user from "root" to the one that is input and executes the
+// SQL statement as that user.
+func execSQLUser(sqlStatement string, user string, errorExpected bool, node int) versionStep {
+	return func(ctx context.Context, t test.Test, u *versionUpgradeTest) {
+		conn, err := u.c.ConnEAsUser(ctx, node, user)
+		require.NoError(t, err)
+		_, err = conn.Exec(sqlStatement)
+		if !errorExpected {
+			require.NoError(t, err)
+		} else {
+			require.Error(t, err)
+		}
+	}
+}
+
+func runRegisterValidateGrantOption(
+	ctx context.Context, t test.Test, c cluster.Cluster, buildVersion version.Version,
+) {
+	const mainVersion = ""
+	predecessorVersion, err := PredecessorVersion(buildVersion)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	u := newVersionUpgradeTest(c,
```

Comment:
Can you add some comments here to make it easier to follow about what we're expecting at each "stage", ie what do we expect to happen before the migration runs, what do we expect to happen after the migration runs.